src/submission/validation-clinical/donor.ts

```diff
@@ -96,6 +96,24 @@ function checkTimeConflictWithSpecimens(
     : [];
 }
 
+const getTreatmentInterval = (clinicalInfo: DeepReadonly<ClinicalInfo>) => {
+  const { treatment_start_interval, treatment_duration } = clinicalInfo;
+  const treatmentInterval =
+    typeof treatment_start_interval === 'number'
+      ? treatment_start_interval + (Number(treatment_duration) || 0)
+      : 0;
+  return treatmentInterval;
+};
+
+const getSpecimenInterval = (clinicalInfo: DeepReadonly<ClinicalInfo>) => {
+  const { specimen_acquisition_interval, specimen_duration } = clinicalInfo;
+  const specimenInterval =
+    typeof specimen_acquisition_interval === 'number'
+      ? specimen_acquisition_interval + (Number(specimen_duration) || 0)
+      : 0;
+  return specimenInterval;
+};
```

Comment:
`ClinicalInfo` has no shape information, update these to accept `Treamtent` or `Specimen` as the generic type. You can pull their clinicalInfo out within the method.